### DISCUSSION TITLE
Find all tonyvr's RP6502 projects on his new GitHub Repository.

### DISCUSSION THREAD

**Discussion by vruumllc** [picocomputer/community#90] [Q&A]
They are all [here](https://github.com/vruumllc).

Enjoy! 

tonyvr

**Comment by vruumllc** [comment]
HI. Has anyone besides n9jcv tried to download and build the projects for cc65 in [my repository](https://github.com/vruumllc)? Were you successful? He is having problems linking to the rp6502 target library, but I can't reproduce it.

Thanks, tonyvr

**Reply by rumbledethumps** [reply]
Be more specific with which repo, which compiler if the repo has both, and the exact error message.

**Reply by vruumllc** [reply]
Hi, some clarifications.
1. He has tried compiling Tetricks, Star Swarms, and Bitmap Graphics Demo using the cc65 compiler, all with similar results.
2. He is getting unresolved symbols for lrand and xreg (i.e. the compiler can't find the rp6502 library)
3. The weird thing is that he can compile the examples project from Rumbledethump's repository just fine.

Here is the message he sent me:

-------------------------------------------

Hello Tony,

I have tried both Tetricks and Star Swarm from your github. When I compile Star Swarms I get these errors.
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:10035: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:6430: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:6502: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:6668: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:6683: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:1101: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:1114: Error: Unresolved external '_lrand'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:9806: Error: Unresolved external '_xreg'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:5963: Error: Unresolved external '_xreg'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/star_swarms.s:1078: Error: Unresolved external '_xreg'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/bitmap_graphics.s:1826: Error: Unresolved external '_xreg'
[build] C:/Users/n9jcv/Desktop/GH/RP6502-Star-Swarms/star_swarms_cc65/src/bitmap_graphics.s:2453: Error: Unresolved external '_xreg'

fyi - I am using VSCODE on Windows.
I get very similar errors, just different line numbers and files from Tetricks. I think maybe an include is missing or something, but I can not seem to figure it out. Any insight to help me?

Thanks for your time and your contributions. I want to learn from Tetricks and Star Swarms that you created.
Bruce

**Reply by rumbledethumps** [reply]
Works for me too. I can reproduce these exact errors with `set(CMAKE_C_FLAGS "--target none")` instead of `set(CMAKE_C_FLAGS "--target rp6502")` but no idea why it's happening.

**Reply by vruumllc** [reply]
Yeah, all I can guess that he has more than one cc65 compiler installed??? I'm stumped too.
Thanks very much for trying it out.  I tested everything when I set up the repository, and I also tried it out on another laptop I have.

**Reply by vruumllc** [reply]
Hi, Rumbledethumps,

We have made some progress trying to compile my stuff under Windows. I think the previous problems were because he was starting vscode from the wrong directory, and so was missing all the magic in the .vscode directory of the projects.

Once we got that all straightened out, he was able to build the Tetricks project ok(!!!), but is still struggling with asset management int the RP6502-Star-Swarms/star_swarms_cc65 project.  It works fine in Ubuntu, so I went and created a Win11 VMware environment and was able to reproduce his problems: 

[build] [rp6502.py] Adding ROM Asset space0.bin.rp6502
[build]     exec_args()
[build]   File "C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py", line 329, in exec_args
[build]     rom.add_rp6502_file(file)
[build]   File "C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py", line 161, in add_rp6502_file
[build]     with open(file, 'rb') as f:
[build]          ^^^^^^^^^^^^^^^^
[build] FileNotFoundError: [Errno 2] No such file or directory: 'space0.bin.rp6502'

Side Note: it bitches about an invalid escape sequence when trying to convert $ to 0x, but I commented that out, and it made no difference.
  
----------------------------------------------------------------------
The full build output is below:

[main] The folder containing the CMake cache is missing. The cache will be regenerated.
[main] Configuring project: star_swarms_cc65 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Release -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_TOOLCHAIN_FILE:FILEPATH=c:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65/tools/rp6502.cmake -SC:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65 -Bc:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- The C compiler identification is unknown
[cmake] -- The ASM compiler identification is unknown
[cmake] -- Found assembler: C:/Users/tonyv/cc65/bin/cl65.exe
[cmake] -- Warning: Did not find file Compiler/-ASM
[cmake] -- Found Python3: C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.12_3.12.752.0_x64__qbz5n2kfra8p0/python3.12.exe (found version "3.12.2") found components: Interpreter
[cmake] -- Configuring done (8.3s)
[cmake] -- Generating done (0.2s)
[cmake] -- Build files have been written to: C:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65/build
[main] Building folder: star_swarms_cc65 all
[build] Starting build
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build c:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65/build --config Release --target all -j 6 --
[build] [  3%] Building C object CMakeFiles/star_swarms.dir/src/bitmap_graphics.c.obj
[build] [  6%] Building C object CMakeFiles/star_swarms.dir/src/star_swarms.c.obj
[build] [ 10%] Linking C executable star_swarms
[build] C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py:183: SyntaxWarning: invalid escape sequence '\$'
[build]   str = re.sub('^\$', '0x', str)
[build] C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py:280: SyntaxWarning: invalid escape sequence '\$'
[build]   str = re.sub('^\$', '0x', str)
[build] Traceback (most recent call last):
[build]   File "C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py", line 347, in <module>
[build] [rp6502.py] Creating C:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65/build/star_swarms.rp6502
[build] [rp6502.py] Adding Binary Asset C:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65/build/star_swarms
[build] [rp6502.py] Adding ROM Asset space0.bin.rp6502
[build]     exec_args()
[build]   File "C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py", line 329, in exec_args
[build]     rom.add_rp6502_file(file)
[build]   File "C:\Users\tonyv\projects\RP6502\RP6502-Star-Swarms\star_swarms_cc65\tools\rp6502.py", line 161, in add_rp6502_file
[build]     with open(file, 'rb') as f:
[build]          ^^^^^^^^^^^^^^^^
[build] FileNotFoundError: [Errno 2] No such file or directory: 'space0.bin.rp6502'
[build] make.exe[2]: *** [star_swarms] Error 1
[build] make.exe[2]: *** Deleting file `star_swarms'
[build] make.exe[1]: *** [CMakeFiles/star_swarms.dir/all] Error 2
[build] make.exe: *** [all] Error 2
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --build c:/Users/tonyv/projects/RP6502/RP6502-Star-Swarms/star_swarms_cc65/build --config Release --target all -j 6 -- exited with code: 2
[driver] Build completed: 00:00:01.578
[build] Build finished with exit code 2
[main] =======================================================
[main] No executable target was found to launch. Please check:
[main]  - Have you called add_executable() in your CMake project?
[main]  - Have you executed a successful CMake configure?
[main] No program will be executed

**Reply by rumbledethumps** [reply]
Try the latest python script from the template files first. You're on an old one if you're getting the escape warning. Then see if you can work the issue some more since you have it reproduced and I don't.

**Reply by vruumllc** [reply]
Will do! Thanks for the tip. I wasn't aware I was out of date.

**Reply by vruumllc** [reply]
Terrific news! -- I copied the latest .vscode and tools directories into the project, and now it works fine under Windows. I will update my repositories.

**Reply by vruumllc** [reply]
OK, I updated all my repos with the latest tools files, then downloaded them and tested that all cc65 projects build and run under both Linux and Windows11.

Thanks!

**Reply by vruumllc** [reply]
I just installed llvm-mos on WIndows11 VM and verified that my builds all work with that too.

**Comment by n9jcv** [comment]
@rumbledethumps @vruumllc 
Thank you both, this has solved my issue and I can now build star swarms in Windows, just download the cc65 version and build.  No other special activities by me, and it built 100%
Thanks to you both!
Bruce
